maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -355,7 +365,8 @@ private ModelSource createStubModelSource( Artifact artifact )
 
         ReactorModelPool modelPool = new ReactorModelPool();
 
-        InternalConfig config = new InternalConfig( request, modelPool );
+        InternalConfig config = new InternalConfig( request, modelPool,
+                useGlobalModelCache() ? getModelCache() : new ReactorModelCache() );
```

Comment:
Why are you passing `new ReactorModelCache()` when you are passing `null` 60 lines above?

Reply:
I'm basically restoring the behavior before changes for MNG-6311 were applied.
I think in case of reading a single project, we don't need a cache as there is no risk that we will read the same pom multiple times; but in the case of using `build(listOfProject,...)`, then the cache is necessary to make sure that poms are read only once (so that 2 children module of the same parent would reference the same parent as MavenProject and not instantiate 2 clones of it).

Reply:
Rechecked the code, the surrouding class does. Isn't then the change above not wrong?

Reply:
Ok, I'll also recheck tomorrow. Thanks for bringing attention there!

Reply:
I think the change is right. See https://github.com/apache/maven/commit/8bc3c207d0aa7cef72171af23d9c4667b2d46c5d (which is the behavior I'd like to avoid when property is set). On lines 294 & 358, it was first `null` then `modelCache`; so I'm really restoring these values when cache is disabled.
I'm not sure it's what's best, but at this point, I'd rather keep focus on fixing the issue rather than investigating more possible improvements.